[dnm] repro for #106108

tbg opened the pull request:
See https://github.com/cockroachdb/cockroach/issues/106108

tbg committed:
repro for https://github.com/cockroachdb/cockroach/issues/106108

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106254)
<!-- Reviewable:end -->

tbg committed:
roachtest: allow cockroach-ephemeral in roachprod-stress

`andrei-jepsen` is very low on quota, and `cockroach-ephemeral` is no
longer used by the nightlies but is well-quotaed, so we should use it
instead.

Remove an old tripwire that tried to prevent folks from using
`cockroach-ephemeral`.

Epic: None
Release note: None

tbg committed:
give deadlocked mutex time to explode

tbg committed:
retiurn nil

tbg commented on pkg/util/syncutil/mutex_sync.go:
```diff
@@ -37,6 +44,45 @@ func (m *Mutex) AssertHeld() {
 type RWMutex struct {
 	sync.RWMutex
+	pcs      []uintptr
+	watching *time.Timer // from mtime.AfterFunc
```
anchor comment that I'll reference from elsewhere
